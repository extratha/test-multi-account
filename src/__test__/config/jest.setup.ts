import '@testing-library/jest-dom';

import mockRouter from "next-router-mock";
import { server } from '../__mocks__/server';
import "jest-canvas-mock";


jest.mock("next/navigation", () => require("next-router-mock"));

jest.mock("next/config", () => () => ({
    setConfig: jest.fn(),
    publicRuntimeConfig: {
        apiToken: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MDY2NzUzNDQsImlhdCI6MTcwNjU4ODk0NCwicHJvZmlsZUlkIjoiZDU1YWY3NDUtOTYyZS00MTk2LTkyOTEtZDg2YTliYWNiZWY2In0.sNgwYjIGLYEVAptqILGgRVGXwt9FCPKHRubEZyaodhg",
        mockBaseApiUrl: "https://www.mock-cariva-health-dashboard.co.th",
        baseApiUrl: "https://www.mock-cariva-health-dashboard.co.th",
        globalEndpoint: "line-telehealth-dashboard",
        publicApiHost: "http://localhost:3000/",
    },
}));

jest.setTimeout(100000);

let routerPushSpy: jest.Spied<typeof mockRouter.push>;

beforeAll(() => {
    server.listen({ onUnhandledRequest: "error" });
});

beforeEach(() => {
    const origPush = mockRouter.push.bind({});
    routerPushSpy = jest.spyOn(mockRouter, "push").mockImplementation(async (...params) => {
        try {
            return await origPush(...params);
        } catch (error) {
            // This error is deliberate to abort Next routing and can be ignored.
            // if (error !== ROUTE_CHANGE_ABORTED) {
            //   throw error;
            // }
        }
        return Promise.resolve(true);
    });
});



afterEach(() => {
    server.resetHandlers();
    routerPushSpy.mockRestore();
});
// Clean up after the tests are finished.
afterAll(() => server.close());
