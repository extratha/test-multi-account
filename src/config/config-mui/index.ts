export { default as RootStyleProvider } from './RootStyleProvider';
export * from './theme';
