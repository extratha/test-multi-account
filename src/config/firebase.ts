// Import the functions you need from the SDKs you need
import { envConfig } from "@/constant/env";
import { initializeApp } from "firebase/app";

// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: envConfig.firebase.apiKey,
  authDomain: envConfig.firebase.authDomain,
  projectId: envConfig.firebase.projectId,
  storageBucket: envConfig.firebase.storageBucket,
  messagingSenderId: envConfig.firebase.messagingSenderId,
  appId: envConfig.firebase.appId,
  measurementId: envConfig.firebase.measurementId,
};

const firebaseApp = initializeApp(firebaseConfig);
export default firebaseApp;
