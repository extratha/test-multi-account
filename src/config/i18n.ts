export const LOCALE_NAMESPACE = "translate";
export const LOCALE_LANGUAGE = {
  TH: "th",
};
