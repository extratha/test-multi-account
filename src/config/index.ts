export const LOCALE_NAMESPACE = "translate";
export const LOCALE_LANGUAGE = {
  TH: "th",
};

export const TOAST_AUTO_HIDE_DURATION = 3000;
export const REMOTE_CONFIG_INTERVAL = 60000;
