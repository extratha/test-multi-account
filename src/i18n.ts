import { notFound } from 'next/navigation';
import { getRequestConfig } from 'next-intl/server';

export const locales = ['th', 'en'];
export const defaultLocale = 'th';
export default getRequestConfig(async (params) => {
  if (!locales.includes(params.locale as string)) {
    return notFound();
  }
  const { locale } = params;
  return {
    messages: {
      ...(await import(`../public/locales/${locale}/common.json`)).default,
    },
  };
});
