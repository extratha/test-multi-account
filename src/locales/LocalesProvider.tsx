"use client";

import { ReactNode, useState } from "react";
import { I18nextProvider } from "react-i18next";

import { initI18next } from "./i18n";
import { LocaleResource } from "@/types/model.ui";

export interface LocalesProviderProps {
  lang: string;
  resource: LocaleResource;
  children: ReactNode;
}

const LocalesProvider = (props: LocalesProviderProps) => {
  const { lang, resource, children } = props;
  const [i18n] = useState(initI18next(lang, resource));

  return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
};

export default LocalesProvider;
