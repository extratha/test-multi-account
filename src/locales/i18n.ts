import i18n from "i18next";
import { initReactI18next } from "react-i18next";

import { LOCALE_LANGUAGE, LOCALE_NAMESPACE } from "@/config/i18n";
import { LocaleResource } from "@/types/model.ui";

export const initI18next = (lang: string, resource: LocaleResource) => {
  i18n.use(initReactI18next).init({
    lng: LOCALE_LANGUAGE.TH,
    fallbackLng: LOCALE_LANGUAGE.TH,
    ns: [LOCALE_NAMESPACE],
    defaultNS: LOCALE_NAMESPACE,
    partialBundledLanguages: false,
    resources: {
      [lang]: { [LOCALE_NAMESPACE]: resource },
    },
  });

  return i18n;
};
