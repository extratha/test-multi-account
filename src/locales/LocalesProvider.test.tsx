import { render } from "@testing-library/react";

import LocalesProvider, { LocalesProviderProps } from "./LocalesProvider";
import { LOCALE_LANGUAGE } from "@/config/i18n";

describe("LocalesProvider", () => {
  let props: LocalesProviderProps;

  beforeEach(() => {
    props = {
      lang: LOCALE_LANGUAGE.TH,
      resource: { text: "Display Text" },
      children: <div data-testid="children" />,
    };
  });

  const renderLocalesProvider = () => {
    return render(<LocalesProvider {...props} />);
  };

  it("should render correctly", () => {
    const { asFragment } = renderLocalesProvider();
    expect(asFragment()).toMatchSnapshot();
  });
});
