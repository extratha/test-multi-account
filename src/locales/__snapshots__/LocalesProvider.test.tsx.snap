// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalesProvider should render correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="children"
  />
</DocumentFragment>
`;
